src/credential_provider.rs

```diff
@@ -300,20 +305,25 @@ impl CredentialProviderBuilder {
         // If endpoint override is not provided, then `c` and `cp` claims must be present.
         let cache_endpoint = cache_endpoint_override
             .or(token_claims.cache_endpoint)
-            .ok_or_else(|| MomentoError::InvalidArgument {
-                description: "auth token is missing cache endpoint and endpoint override is missing. One or the other must be provided".into(),
-                source: None,
-            })?;
+            .ok_or_else(|| MomentoError::InvalidArgument(SdkError {
+                message: "auth token is missing cache endpoint and endpoint override is missing. One or the other must be provided".into(),
+                error_code: MomentoErrorCode::InvalidArgumentError,
+                inner_exception: None,
+                details: None
+            }));
+
         let control_endpoint = control_endpoint_override
             .or(token_claims.control_endpoint)
-            .ok_or_else(|| MomentoError::InvalidArgument {
-                description: "auth token is missing control endpoint and endpoint override is missing. One or the other must be provided.".into(),
-                source: None,
-            })?;
+            .ok_or_else(|| MomentoError::InvalidArgument(SdkError {
+                message: "auth token is missing control endpoint and endpoint override is missing. One or the other must be provided.".into(),
+                error_code: MomentoErrorCode::InvalidArgumentError,
+                inner_exception: None,
+                details: None
+            }));
         Ok(CredentialProvider {
             auth_token,
-            cache_endpoint: CredentialProviderBuilder::https_endpoint(cache_endpoint),
-            control_endpoint: CredentialProviderBuilder::https_endpoint(control_endpoint),
+            cache_endpoint: CredentialProviderBuilder::https_endpoint(cache_endpoint?),
+            control_endpoint: CredentialProviderBuilder::https_endpoint(control_endpoint?),
```

Comment:
code hint suggested using `?` here but I'm not entirely sure yet this is correct

Reply:
TIL: https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html